src/nft/hooks/useSendTransaction.ts

```diff
@@ -36,21 +37,19 @@ export const useSendTransaction = create<TxState>()(
       purchasedWithErc20: false,
       clearTxHash: () => set({ txHash: '' }),
       setState: (newState) => set(() => ({ state: newState })),
-      sendTransaction: async (signer, selectedAssets, transactionData, purchasedWithErc20) => {
-        const address = await signer.getAddress()
+      sendTransaction: async (provider, selectedAssets, transactionData, purchasedWithErc20) => {
+        const address = await provider.getSigner().getAddress()
         try {
-          const txNoGasLimit = {
+          set({ state: TxStateType.Signing })
+
+          // TODO(signer should be provider)
```

Comment:
This was a note to myself that I'd resolved and forgotten to remove: literally to replace the signer on L39/40 with a provider, because the new method uses a provider.